Changelog 186 (#172)
intellij-rust/intellij-rust.github.io: _posts/2023-01-16-changelog-186.markdown

---
layout: post
title: "IntelliJ Rust Changelog #186"
date: 2023-01-16 13:00:00 +0300
---

## New Features

* [#9747] Support `Create field` quick-fix for struct literals with `Self`

  {% include gif-img.html path="/assets/posts/changelog-186/create-field" w="700px" %}

* [#9794] Remove unused imports inside functions by `Optimize import` action

>   if [Unused imports](https://github.com/intellij-rust/intellij-rust/issues/2158) inspections is enabled

Reviewer: ```suggestion
  (only if [Unused imports](https://github.com/intellij-rust/intellij-rust/issues/2158) inspection is enabled)
```

  {% include gif-img.html path="/assets/posts/changelog-186/unused-imports-in-function" w="700px" %}

* [#9441] Fully qualify paths from the local crate that cannot be imported after paste.
  Note, the corresponding feature is disabled by default.
  You can turn it on via `Rust | Insert imports on paste` option in `Preferences | Editor | General | Auto Import` panel (by [@Kobzol])

  {% include gif-img.html path="/assets/posts/changelog-186/import-on-paste" w="700px" %}

* [#9883], [#9899] Annotate [exclusive range patterns](https://github.com/rust-lang/rust/issues/37854) as experimental

## Fixes

* [#9928] Improve error messages for already defined names:
    - Produce proper [E0403](https://doc.rust-lang.org/error_codes/E0403.html) error for const generic parameters with duplicate names

>     - Update error messages for already defined names to provide the same error message as the compiler

Reviewer: ```suggestion
    -  Use the same error messages as the Rust compiler
```
